src/utils/src/kernel_version.rs

```diff
@@ -6,12 +6,16 @@ use std::result::Result;
 
 use libc::{uname, utsname};
 
-#[derive(Debug, derive_more::From)]
+#[derive(Debug, thiserror::Error, displaydoc::Display)]
 pub enum Error {
-    Uname(IoError),
-    InvalidUtf8(std::string::FromUtf8Error),
+    /// Uname: {0}
+    Uname(#[from] IoError),
+    /// InvalidUtf8: {0}
+    InvalidUtf8(#[from] std::string::FromUtf8Error),
+    /// InvalidFormat
```

Comment:
nit: Can we be a bit more explicit about what the "invalid format" here is?

Reply:
fixed